### PR TITLE
Refact records

### PR DESCRIPTION
**Pull request by theengineear**
Woo! After some of the recent refacts, this change became much easier. I nested `records` at `state/old`.

I think we still need to think about naming and what we want to expose as a top-level export. However, we're getting a lot closer to the organization we're ultimately looking for in `/core`.

### CODE REVIEW

**Comment by theengineear** [comment]
@rgbkrk A little overdue, but here you go 😅.

File: applications/desktop/__tests__/renderer/native-window-spec.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { remote } from "electron";
 import { from } from "rxjs/observable/from";
 
 import * as nativeWindow from "../../src/notebook/native-window";
-import { makeAppRecord, makeDocumentRecord } from "@nteract/core/records";
+import { state as stateModule } from "@nteract/core";
```
Comment (rgbkrk) [inline applications/desktop/__tests__/renderer/native-window-spec.js:7]: It would be really tempting to call the state module "core".

**Review by rgbkrk — APPROVED** [review]
Looking super great here!

File: applications/desktop/__tests__/renderer/native-window-spec.js (hunk 1 above)
Comment (theengineear) [reply applications/desktop/__tests__/renderer/native-window-spec.js:7]: Heh, yah that's true. We should figure out the right names for all this stuff soon. I like that all this is under `/state` though, it makes it really easy to manage in the repo.

**Comment by theengineear** [comment]
Thanks for the quick review!

File: applications/desktop/__tests__/renderer/native-window-spec.js (hunk 1 above)
Comment (rgbkrk) [reply applications/desktop/__tests__/renderer/native-window-spec.js:7]: Yeah my only worry is the super common use of the variable named state.

Comment (theengineear) [reply applications/desktop/__tests__/renderer/native-window-spec.js:7]: Yup, definitely worthwhile to not block that variable name 😆.

**Pull request merged**